fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/HeadersRedactor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Armory, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.fiat.shared;
+
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+class HeadersRedactor {
+
+  private static final String SECRET_DATA_HEADER_FORMAT = "X-";
+  private static final String AUTHORIZATION_HEADER = "Authorization";
+  private static final String SECRET_DATA_VALUE = "**REDACTED**";
+
+  public Map<String, String> getRedactedHeaders(HttpServletRequest request) {
+    Map<String, String> headers = new HashMap<>();
+
+    if (request.getHeaderNames() != null) {
+      for (Enumeration<String> h = request.getHeaderNames(); h.hasMoreElements(); ) {
+        String headerName = h.nextElement();
+        String headerValue = getRedactedHeaderValue(headerName, request.getHeader(headerName));
+        headers.put(headerName, headerValue);
+      }
+    }
+    return headers;
+  }
+
+  private static String getRedactedHeaderValue(String headerName, String headerValue) {
```

Comment:
IF I'm following the logic here correctly, this means that X-* headers would have their values masked.  As a rule, anything with X- should be included and everything else masked/not included.  For example, X-Spinnaker-User is a KEY header to display as that identifies the user who made a request.  This ALSO simplifies the rather complicated "Look for the header... then look at the header starting again later on" logic.  

Instead I'd recommend something like:
"IF starts with X-, keep it and include the value.  Otherwise, don't even include it masked" 
as probably a safer setup.

Reply:
@jasonmcintosh wouldn't it make sense to keep the non X- headers with masked values for debugging purposes? I am not saying we need to keep em, but their presence shouldn't hurt us and they can help when verifying some issues.